### PR TITLE
API for emailLogEntry 

### PR DESCRIPTION
**Pull request by taslangraham**
For pkp/pkp-lib#10412

### CODE REVIEW

File: api/v1/emails/PKPEmailController.php [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        return parent::authorize($request, $args, $roleAssignments);
+    }
+
+    /**
+     * Get a collection of emails for an author
+     */
+    protected function getMany(Request $illuminateRequest): JsonResponse
+    {
+        $allowedQueryParams = ['submissionId', 'eventType'];
+        $authorEmailLogEventTypes = [ SubmissionEmailLogEventType::EDITOR_NOTIFY_AUTHOR, SubmissionEmailLogEventType::COPYEDIT_NOTIFY_AUTHOR,SubmissionEmailLogEventType::PROOFREAD_NOTIFY_AUTHOR];
```
Comment (ewhanson) [inline api/v1/emails/PKPEmailController.php:95]: Some spacing issues in the array.

File: api/v1/reviews/PKPReviewController.php [hunk 2]
```diff
@@ -131,7 +131,7 @@ public function getHistory(Request $illuminateRequest): JsonResponse
         $declineEmail = null;
         if ($reviewAssignment->getDeclined()) {
             $emailLogs = EmailLogEntry::withAssocId($submissionId)
-                ->withEventType(SubmissionEmailLogEventType::REVIEW_DECLINE)
+                ->withEventTypes([SubmissionEmailLogEventType::REVIEW_DECLINE])
```
Comment (ewhanson) [inline api/v1/reviews/PKPReviewController.php:134]: Just wanting to make sure you checked the OJS/OMP/OPS repos for any place this method appears to make sure it's changed there as well.

**Review by ewhanson — CHANGES_REQUESTED** [review]

File: api/v1/reviews/PKPReviewController.php (hunk 2 above)
Comment (taslangraham) [reply api/v1/reviews/PKPReviewController.php:134]: Yes, I did check. This is the only place where the method is being used.

File: api/v1/emails/PKPEmailController.php [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+        $currentUser = $this->getRequest()->getUser();
+
+        foreach ($illuminateRequest->query->keys() as $queryParam) {
+            if (!in_array($queryParam, $allowedQueryParams)) {
+                return response()->json([
```
[the remark this answers is not in the document]
Comment (taslangraham) [reply api/v1/emails/PKPEmailController.php:101]: Noted. I'll update accordingly

**Pull request merged**